backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/ModularContractService.java

```diff
@@ -464,4 +459,63 @@ public void closeContractsOnOrderCloseIfNeededAndAllowed(@NonNull final I_C_Orde
 
 		modularAndInterimContracts.forEach(contractChangeBL::endContract);
 	}
+
+	private Set<I_C_Flatrate_Term> getModularAndInterimContracts(@NonNull final OrderId orderId)
+	{
+		return flatrateBL.getByOrderId(orderId).stream()
+				.filter(this::isModularOrInterimContract)
+				.collect(Collectors.toSet());
+	}
+
+	private Set<OrderId> getSalesOrderIdsLinkedToPurchaseModularContract(@NonNull final Set<FlatrateTermId> modularContractIds)
+	{
+		return orderBL.streamOrderLines(OrderLineQuery.builder().modularPurchaseContractIds(modularContractIds).build())
+				.map(I_C_OrderLine::getC_Order_ID)
+				.map(OrderId::ofRepoId)
+				.collect(Collectors.toSet());
+	}
+
+	public boolean isModularOrder(@NonNull final OrderId orderId)
+	{
+		return !getModularAndInterimContracts(orderId).isEmpty();
+	}
+
+	public boolean hasLinkedPurchaseModularContracts(@NonNull final OrderId salesOrderId)
+	{
+		return !getOrderLinesWithLinkedPurchaseModularContracts(salesOrderId).isEmpty();
```

Comment:
wondering if it makes sense to introduce & use orderBL.anyMatch(OrderLineQuery),
and then u could have here some method: OrderLineQuery purchaseModularContract(salesOrderId)